src/common/internal/test_group.ts

```diff
@@ -446,40 +434,66 @@ class RunCaseSpecific implements RunCase {
       return didSeeFail ? 'fail' : 'pass';
     };
 
-    rec.start();
-    if (this.subcases) {
-      let totalCount = 0;
-      let skipCount = 0;
-      for (const subParams of this.subcases) {
-        rec.info(new Error('subcase: ' + stringifyPublicParams(subParams)));
-        try {
-          const params = mergeParams(this.params, subParams);
-          const subcaseQuery = new TestQuerySingleCase(
-            selfQuery.suite,
-            selfQuery.filePathParts,
-            selfQuery.testPathParts,
-            params
-          );
-          await this.runTest(rec, params, true, getExpectedStatus(subcaseQuery));
-        } catch (ex) {
-          if (ex instanceof SkipTestCase) {
-            // Convert SkipTestCase to info messages
-            ex.message = 'subcase skipped: ' + ex.message;
-            rec.info(ex);
-            ++skipCount;
-          } else {
-            // Since we are catching all error inside runTest(), this should never happen
-            rec.threw(ex);
+    try {
+      rec.start();
+      const inst = new this.fixture(rec, this.params);
```

Comment:
Here we are reusing the same fixture many times. Don't we have a concern that tests could be storing state on the fixture that would break stuff because it is not expected?

Reply:
Yea, definitely. I've been working on a bigger refactor to let the fixture class perform some setup operations whose results then get passed into the fixture constructor.

It restructures things quite a bit so I'm still working out the details and making it not look like a mess.